code/__HELPERS/icons.dm

```diff
@@ -943,6 +943,22 @@ var/global/list/friendly_animal_types = list()
 		return J
 	return 0
 
+/atom/proc/cut_overlays()
+	overlays.Cut()
+	overlays += priority_overlays
+
+/atom/proc/add_overlay(image, priority = 0)
+	if(image in overlays)
+		return
+	var/list/new_overlays = overlays.Copy()
+	if(priority)
+		priority_overlays += image
+		new_overlays += image
+	else
+		new_overlays -= priority_overlays
+		new_overlays += image
+		new_overlays += priority_overlays
+	overlays = new_overlays

```

Comment:
Yes it does.
On Jun 4, 2016 7:30 AM, "xxalpha" <notifications@github.com> wrote:

> In code/__HELPERS/icons.dm
> <https://github.com/tgstation/tgstation/pull/18225#discussion_r65803224>:
>
> > +/atom/proc/cut_overlays()
> > +	overlays.Cut()
> > +	overlays += priority_overlays
> > +
> > +/atom/proc/add_overlay(image, priority = 0)
> > +	if(image in overlays)
> > +		return
> > +	var/list/new_overlays = overlays.Copy()
> > +	if(priority)
> > +		priority_overlays += image
> > +		new_overlays += image
> > +	else
> > +		new_overlays -= priority_overlays
> > +		new_overlays += image
> > +		new_overlays += priority_overlays
> > +	overlays = new_overlays
>
> Does this system take into account appearances? If I add an object to
> new_overlays does doing overlays = new_overlays use that object's
> appearance?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/18225/files/d69b6ac8d844c752450b8a107dc7c3d5a1581205#r65803224>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvgJX3qiJC56vdciVWQPKh8r2Zh-iuWks5qIYwTgaJpZM4It7Wu>
> .
>